--- a/drivers/infiniband/hw/cxgb3/Makefile	2007-09-12 05:28:57.000000000 -0500
+++ b/drivers/infiniband/hw/cxgb3/Makefile	2007-09-26 09:59:53.000000000 -0500
@@ -2,10 +2,13 @@ EXTRA_CFLAGS += -I$(TOPDIR)/drivers/net/
 		-I$(TOPDIR)/drivers/infiniband/hw/cxgb3/core
 
 obj-$(CONFIG_INFINIBAND_CXGB3) += iw_cxgb3.o
+obj-$(CONFIG_INFINIBAND_CXGB3) += rdma_krping.o
 
 iw_cxgb3-y :=  iwch_cm.o iwch_ev.o iwch_cq.o iwch_qp.o iwch_mem.o \
 	       iwch_provider.o iwch.o cxio_hal.o cxio_resource.o
 
+rdma_krping-y := krping.o getopt.o
+
 ifdef CONFIG_INFINIBAND_CXGB3_DEBUG
 EXTRA_CFLAGS += -DDEBUG
 endif
